docs/rholang/rholangtut.md

```diff
@@ -0,0 +1,501 @@
+
+3) We create a `port` channel for interacting with the coat check as well as a `table` name which will be used in storing/retrieving values in the coat check.
+
+4) We send `port` out to the caller, so they can interact with the coat check.
+
+5, 11, 17) We define different methods which can be called by sending a message on `port`. This is done by specifying mutually exclusive patterns the message on `port` can match, with the first element of the message being the method name and subsequent elements being the argument(s) and return channel. Using the `<=` arrow instead of the `<-` arrow means that the `for`s are "replicated". This gives them the same behavior as `contract`s, i.e. the process listening for messages on `port` persists after spawning an instance of its body.
+
+8) We take advantage of being able to quote any process to make a name in order to create a unique name for each value to be stored at. The process `*ticket | *table` is produced by the concurrent composition of the processes produced by unquoting the names `ticket` and `table`. That process can then be quoted to form a unique name that is then used to store the value by sending it on the name.
+
+## Dining philosophers and deadlock
+
+     1 new philosopher1, philosopher2, north, south, knife, spoon in {
+     2   north!(*knife) |
+     3   south!(*spoon) |
+     4   for (@knf <- north) { for (@spn <- south) {
+     5     philosopher1!("Complete!") |
+     6     north!(knf) |
+     7     south!(spn)
+     8   } } |
+     9   for (@spn <- south) { for (@knf <- north) {
+    10     philosopher2!("Complete!") |
+    11     north!(knf) |
+    12     south!(spn)
+    13   } }
+    14 }
+
+The dining philosophers problem has two philosophers that share only one set of silverware.  Philosopher1 sits on the east side of the table while Philosopher2 sits on the west. Each needs both a knife and a spoon in order to eat.  Each one refuses to relinquish a utensil until he has used both to take a bite.  If both philosophers reach first for the utensil at their right, both will starve: Philosopher1 gets the knife, Philosopher2 gets the spoon, and neither ever lets go.
+
+Here's how to solve the problem:
+
+     1 new philosopher1, philosopher2, north, south, knife, spoon in {
+     2   north!(*knife) |
+     3   south!(*spoon) |
+     4   for (@knf <- north; @spn <- south) {
+     5     philosopher1!("Complete!") |
+     6     north!(knf) |
+     7     south!(spn)
+     8   } |
+     9   for (@spn <- south; @knf <- north) {
+    10     philosopher2!("Complete!") |
+    11     north!(knf) |
+    12     south!(spn)
+    13   }
+    14 }
+
+4, 9) The join operator, denoted with a semicolon `;`, declares that the continuation should only proceed if there is a message available on each of the channels simultaneously, preventing the deadlock above.
+
+## Crypto channels
+
+### Hashing
+
+There are three hashing functions available:
+- keccak256
```

Comment:
Some markdown renderers need a blank line before the dash to trigger the unordered list. Same thing online 272.

Reply:
By "some" you mean not github because it renders properly here.